### DISCUSSION TITLE
additionalProp1 now showing on all auto-generated examples in openapi after upgrade to 0.99

### DISCUSSION THREAD

**Discussion by FISHMANPET** [fastapi/fastapi#9820] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, Extra

class Hero(BaseModel, extra=Extra.forbid):
    name: str

class Team(BaseModel):
    name: str

app = FastAPI()

@app.get("/hero")
def get_hero(hero: Hero) -> Hero:
    return hero

@app.get("/team")
def get_team(team: Team) -> Team:
    return team
```

### Description

When moving from 0.98 (or below) to 0.99 (.0 or .1) my auto-generated examples all have an added `"additionalProp1": {}` key, despite all of my models having extras set to forbidden.

Looking at the generated openapi.json of a previous version of my app and my local copy running the latest FastAPI, it appears they both have `"additionalProperties": false` set on the schemas, so there doesn't appear to be a change there.

Since this coincides with the move to OpenAPI 3.10 but also there were some issues between 0.99.0 and 0.99.1 around additionalProperties, I'm not sure if this is a bug in FastAPI, a bug in OpenAPI, or just some new default that's not documented.

I can't find any clear documentation from OpenAPI of how to disable showing AdditionalProp1 on the models, much less how I would pass that configuration to OpenAPI via FastAPI.

In the given example, the Hero model has `"additionalProperties": false` set in the openapi schema, and this auto-generated example:
```
{
  "name": "string",
  "additionalProp1": {}
}
```
Whereas the Team model doesn't have an `additionalProperties` setting, and the auto-generated example looks like:
```
{
  "name": "string"
}
```

How can I prevent this additionalProp1 from showing up in my example?

### Operating System

Windows

### FastAPI Version

0.99.1

### Python Version

3.11

**Comment by eugenelandau** [comment]
This problem still persisted with FastAPI 0.100.0. I only managed to solve it by setting `json_schema_extra` in `model_config`:

```python
class SomeModel(BaseModel, extra=Extra.forbid):
    some_field: str

    model_config = ConfigDict(json_schema_extra={
        'example': {
            'some_field': 'some_field',
        },
    })
```

But there has to be a more general solution..

**Comment by necromeo** [comment]
I've digged a bit and played around with FastAPI's source code. The issue comes from bumping up the openapi version from 3.0.2 to 3.1.0 [starting from FastAPI 0.99.0](https://github.com/tiangolo/fastapi/commit/7dad5a820bfe99e49a6cfaefde537e09644c2c2b).
Given that 0.99.1 is the last version with Pydantic V1, @tiangolo would it be possible to revert using 3.0.2 or 3.0.3 (which from my findings works as expected)?
At least on my team, we are looking forward to migrating to 0.100 but it won't be anytime soon. Also, I have no idea whether this same issue is present in said version.

**Reply by necromeo** [reply]
Found a way to fix it. I sent a PR but closed it as there were WAY too many differences between 0.99.1 and the master branch.
If you have to stick to 0.99.1 like myself for a while, take a look at the closed PR https://github.com/tiangolo/fastapi/pull/9886.
I know I'll be using my fork until I can move over to 0.100.

**Reply by tiangolo** [reply]
Just in case, the latest version of FastAPI (currently 0.103.0) still supports Pydantic v1, and it will keep supporting it for a while, hopefully enough for everyone (or most?) to migrate to Pydantic v2.

**Comment by adavis444** [comment, marked as answer]
This is a bug in swagger-ui for the OpenAPI Specification 3.1.0 (Issue: https://github.com/swagger-api/swagger-ui/issues/9022) that would be resolved by https://github.com/swagger-api/swagger-ui/pull/9023.

In the meantime you could consider using `/redoc` instead of `/docs`.

**Reply by Kludex** [reply]
Thanks @adavis444 👍

**Reply by FISHMANPET** [reply]
Thanks to @adavis444's PR, [Swagger UI 5.2.0](https://github.com/swagger-api/swagger-ui/releases/tag/v5.2.0) has been released.  I've tested my app with the latest FastAPI (0.100.0) and the `additionalProp1` keys are no longer present in the Swagger UI.  Thanks for fixing this @adavis444!

**Comment by AndrewLaganaro** [comment]
FastAPI 0.100.0 and it still happens. Had to reffer to:

https://github.com/tiangolo/fastapi/issues/5470#issuecomment-1273472764

```python
class CompanyDataOut(BaseModel):
    __root__: dict[City, Misc] = Field(example="""
{
  "city1": {
    "content": {}
  },
  "city2": {
    "content": {}
  },
  "city3": {
    "content": {}
  }
}""")
```

on my case:

```python
class ItemBatch(BaseModel):
    items: Dict[str, Item] = Field(example="""
{
    "item1": {
      "id": 0,
      "name": "string",
      "description": "string",
      "price": 0
    },
    "item2": {
      "id": 0,
      "name": "string",
      "description": "string",
      "price": 0
    },
    "item3": {
      "id": 0,
      "name": "string",
      "description": "string",
      "price": 0
    }
}""")
```

which solved my issue.
